ios/keyman/Keyman/Keyman/MainViewController.swift

```diff
@@ -0,0 +1,1200 @@
+    if launchUrl != nil {
+      perform(#selector(self.dismissActivityIndicator), with: nil, afterDelay: 1.0)
+      let error = notification.userInfo?[NSUnderlyingErrorKey] as! Error
+      showAlert(withTitle: "Keyboard Download Error", message: error.localizedDescription,
+                cancelButtonTitle: "OK", otherButtonTitles: nil, tag: -1)
+      launchUrl = nil
+    }
+  }
+
+  func languagesUpdated(_ notification: Notification) {
+    updateStatus = 1
+  }
+
+  func languagesDownloadFailed(_ notification: Notification) {
+    updateStatus = -1
+  }
+
+  func keyboardPickerDismissed(_ notification: Notification) {
+    textView.becomeFirstResponder()
+    if UIDevice.current.userInterfaceIdiom == .pad {
+      perform(#selector(self.showGetStartedView), with: nil, afterDelay: 1.0)
+    }
+  }
+
+  func launched(fromUrl notification: Notification) {
+    if let url = notification.userInfo?[urlKey] as? URL, url.query != nil {
+      launchUrl = url
+      if updateStatus > 0 {
+        performAction(from: url)
+      }
+    } else {
+      launchUrl = nil
+    }
+  }
+
+  func keyboardRemoved(_ notification: Notification) {
+    guard let kbInfo = notification.userInfo?[kKeymanKeyboardInfoKey] as? [String: Any] else {
+      return
+    }
+    guard let font = kbInfo[kKeymanFontKey] as? String else {
+      return
+    }
+    guard let profile = profileNameWithJSONFont(font) else {
+      return
+    }
+    profiles2Install = profiles2Install.filter { $0 != profile }
+    profiles2Install.append(profile)
+    checkedProfiles = checkedProfiles.filter { $0 != profile }
+    let userData = AppDelegate.activeUserDefaults()
+    userData.set(checkedProfiles, forKey: checkedProfilesKey)
+    userData.synchronize()
+  }
+
+  func infoButtonClick(_ sender: Any?) {
+    UIView.setAnimationDelegate(self)
+    let animSelector = NSSelectorFromString("animationDidStop:finished:context:")
+    UIView.setAnimationDidStop(animSelector)
+    UIView.beginAnimations(nil, context: nil)
+    UIView.setAnimationDuration(0.75)
+    UIView.setAnimationTransition((!textView.isHidden ? .flipFromRight : .flipFromLeft), for: view, cache: true)
+
+    if !textView.isHidden {
+      textView.isHidden = true
+      infoView.view.isHidden = false
+      infoView.reloadKeymanHelp()
+    } else {
+      textView.isHidden = false
+      infoView.view.isHidden = true
+    }
+
+    UIView.commitAnimations()
+
+    if !textView.isHidden {
+      navigationItem.titleView = nil
+      navigationItem.rightBarButtonItem = nil
+      setNavBarButtons()
+      if wasKeyboardVisible {
+        perform(#selector(self.showKeyboard), with: nil, afterDelay: 0.75)
+      }
+      if shouldShowGetStarted() {
+        perform(#selector(self.showGetStartedView), with: nil, afterDelay: 0.75)
+      }
+    } else {
+      _ = dismissDropDownMenu()
+      popover?.dismiss(animated: false)
+      wasKeyboardVisible = textView.isFirstResponder
```

Comment:
Another one of those `actionSheet dismissWithClicked`... lines disappeared here.